src/main/java/seedu/address/model/client/insurance/InsurancePlansManager.java

```diff
@@ -109,7 +114,7 @@ public void deletePlan(InsurancePlan plan) {
      */
     public void checkIfPlanOwned(InsurancePlan plan) throws InsurancePlanException {
         for (InsurancePlan p : insurancePlans) {
-            if (p.equals(plan)) {
+            if (p.getInsurancePlanId() == plan.getInsurancePlanId()) {
```

Comment:
was this modified because the new equals() method in InsurancePlan tests for equality in claims too?

Reply:
Yep seems like it!

Reply:
Yeps, it led to a bug where insurance plan is not detected although the insurance plan was already added to the client